api/src/test/java/org/apache/iceberg/TestSchema.java

```diff
@@ -95,14 +140,21 @@ public void testUnsupportedInitialDefault(int formatVersion) {
             formatVersion);
   }
 
-  @Test
-  public void testSupportedInitialDefault() {
-    assertThatCode(() -> Schema.checkCompatibility(INITIAL_DEFAULT_SCHEMA, 3))
+  private static int[] testSupportedInitialDefault =
+      IntStream.rangeClosed(DEFAULT_VALUES_MIN_FORMAT_VERSION, MAX_FORMAT_VERSION).toArray();
+
+  @ParameterizedTest
+  @FieldSource
+  public void testSupportedInitialDefault(int formatVersion) {
+    assertThatCode(() -> Schema.checkCompatibility(INITIAL_DEFAULT_SCHEMA, formatVersion))
         .doesNotThrowAnyException();
   }
 
+  private static int[] testSupportedWriteDefault =
+      IntStream.rangeClosed(1, MAX_FORMAT_VERSION).toArray();
+
   @ParameterizedTest
-  @ValueSource(ints = {1, 2, 3})
+  @FieldSource
```

Comment:
```suggestion
  @FieldSource(value = "supportedWriteDefault")
```
this makes it clearer which field parameters are actually being used. I would also update all the others

Reply:
That's fine, I kind of like how it just uses the exact same name though